addons/mrp/models/mrp_production.py

```diff
@@ -486,6 +486,19 @@ def _compute_picking_ids(self):
                 ('group_id', '=', order.procurement_group_id.id), ('group_id', '!=', False),
             ])
             order.picking_ids |= order.move_raw_ids.move_orig_ids.picking_id
+            # Count only those pickings related to the current MO in a 2-step or 3-step MRP process.
+            if order.warehouse_id.manufacture_steps in ['pbm_sam', 'pbm']:
+                # We need to use `picking.production_ids` because if MO or picking cancels, we can't decide from which MO picking is created.
+                filtered_picking_ids = order.picking_ids.filtered(
+                    lambda picking: any(
+                        order in (picking.production_ids |
+                                move.move_dest_ids.raw_material_production_id |
+                                move.move_orig_ids.production_id)
+                        for move in picking.move_ids
+                    )
+                )
+                # Unlink unrelated pickings from current MO.
+                order.picking_ids = [(3, picking.id, False) for picking in (order.picking_ids - filtered_picking_ids)]
```

Comment:
```suggestion
                order.picking_ids = [Command.unlink(picking.id) for picking in (order.picking_ids - filtered_picking_ids)]
```
It's easier to understand Command than just a number '3'